pandas/core/indexes/timedeltas.py

```diff
@@ -65,6 +65,14 @@ class TimedeltaIndex(DatetimeTimedeltaMixin):
         Make a copy of input ndarray.
     name : object
         Name to be stored in the index.
+    dtype : numpy.dtype
+        Currently, only ``numpy.dtype("timedelta64[ns]")`` is accepted.
+
+         .. warning::
+
+           A future version of pandas will change dtype to be an instance
```

Comment:
Not familiar with that are of the code. Is this already enforced?

Reply:
Yes, _dtype_ is enforced to be _'timedelta64[ns]'_ inside the constructor `TimedeltaArray._from_sequence_not_strict`, which is called inside [TimedeltaIndex constructor](https://github.com/pandas-dev/pandas/blob/main/pandas/core/indexes/timedeltas.py#L161). The definition of `TimedeltaArray._from_sequence_not_strict` is [here](https://github.com/pandas-dev/pandas/blob/main/pandas/core/arrays/timedeltas.py#L231)

Reply:
So the warning is no longer necessary?

Reply:
I don't know, since the instantiated class [TimedeltaArray](https://pandas.pydata.org/docs/dev/reference/api/pandas.arrays.TimedeltaArray.html) reports that warning, I've copied it. Do I remove it?

Reply:
Can you show an example? Code-example

Reply:
That shows the warning you are describing here

Reply:
Ok, I got the question. The `TimedeltaArray` docstring reports this warning (see here https://pandas.pydata.org/docs/dev/reference/api/pandas.arrays.TimedeltaArray.html) and also the `TimedeltaArray.dtype` property docstring report a similar warning https://github.com/pandas-dev/pandas/blob/main/pandas/core/arrays/timedeltas.py#L160.

This is the function that validate the dtype of `TimedeltaArray` https://github.com/pandas-dev/pandas/blob/main/pandas/core/arrays/timedeltas.py#L1029

Reply:
@phofl 
is there something that I can do to get it merged? thank you